test/integration/features/basic.feature

```diff
@@ -10,6 +10,61 @@ Feature: Basic
      Then Minishift should have state "Running"
       And Minishift should have a valid IP address
 
+  Scenario: User is able to do ssh into Minishift VM
+    Given Minishift has state "Running"
+     When executing "minishift ssh echo hello"
+     Then stderr should be empty
+      And exitcode should equal 0
+      And stdout should contain
+      """
+      hello
+      """
+
+  Scenario: User is able to set custom Docker specific environment variables
+    Given Minishift has state "Running"
+     When executing "minishift ssh cat /var/lib/boot2docker/profile"
+     Then stderr should be empty
+      And exitcode should equal 0
+      And stdout should contain
+      """
+      export "FOO=BAR"
+      export "BAZ=BAT"
+      """
+
+  Scenario Outline: User can set, get, view and unset values in configuration file
+    User is able to setup persistent configuration of Minishift using "config" command
+    and its subcommands.
+    Given Minishift has state "Running"
+     When executing "minishift config set <property> <value>"
+     Then JSON config file contains key "<property>" with value "<value>"
+      And stdout of command "minishift config get <property>" is equal to "<value>"
+      And stdout of command "minishift config view --format {{.ConfigKey}}:{{.ConfigValue}}" contains "<property>:<value>"
+     When executing "minishift config unset <property>"
+     Then stdout of command "minishift config get <property>" is equal to "<nil>"
+      And JSON config file does not contain key "<property>"
+      
+  Examples: Config values to work with
+    | property  | value |
+    | disk-size | 22g   |
+    | memory    | 2222  |
+    | cpus      | 3     |
+
+  Scenario: Commands console and ip returns valid ip
+    User is able to get ip of Minishift VM with command "ip" and url adress of OpenShift instance
+    by running command "console" with flag "--url". Both commands should contain same IP.
+    Given Minishift has state "Running"
+     When executing "minishift console --url"
+     Then stdout should be valid address
+      And "minishift ip" should return same ip
+
+  Scenario: OpenShift is ready after startup
+    After startup of Minishift OpenShift instance should respond correctly on its html endpoints
+    and OpenShift web console should be accessible.
+    Given Minishift has state "Running"
+     Then response of OpenShift at "/healthz" contains "ok"
```

Comment:
"Then body of HTTP request to OpenShift at "/healthz" contains "ok"

or we could make it even more generic and introduce some variables which get interpolated. Then we can have a generic step like:

    "Then body of HTTP request "https://#{vm_ip}:8443/healthz" contains "ok", 

but also 

    "Then body of HTTP request to "https://myapp.#{routing-suffix}/foo" contains "Welcome to me app"

What I am suggesting here is to provide a syntax as well as a predefined set of known variables which can be used for this and maybe other commands. Variables which then in the step definition implementation get replaced with the actual dynamic values. Without this type of mechanism we will have a lot of problems writing tests around things which depend on for example the routing suffix or the IP.